test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -361,96 +388,68 @@ public final void testUnknownObjectException() throws IOException {
      *     }
      * }
      */
-    private void unknownObjectExceptionTest(String validQuery) throws IOException {
-        //TODO building json by concatenating strings makes the code unmaintainable, we should rewrite this test
-        assertThat(validQuery, containsString("{"));
-        int level = 0;
-        //track whether we are within quotes as we may have randomly generated strings containing curly brackets
-        boolean withinQuotes = false;
-        boolean expectedException = true;
-        int objectHoldingArbitraryContentLevel = 0;
-        for (int insertionPosition = 0; insertionPosition < validQuery.length(); insertionPosition++) {
-            if (validQuery.charAt(insertionPosition) == '"') {
-                withinQuotes = withinQuotes == false;
-            } else if (withinQuotes == false && validQuery.charAt(insertionPosition) == '}') {
-                level--;
-                if (expectedException == false) {
-                    //track where we are within the object that holds arbitrary content
-                    objectHoldingArbitraryContentLevel--;
-                }
-                if (objectHoldingArbitraryContentLevel == 0) {
-                    //reset the flag once we have traversed the whole object that holds arbitrary content
-                    expectedException = true;
-                }
-            } else if (withinQuotes == false && validQuery.charAt(insertionPosition) == '{') {
-                //keep track of which level we are within the json so that we can properly close the additional object
-                level++;
-                //if we don't expect an exception, it means that we are within an object that can contain arbitrary content.
-                //in that case we ignore the whole object including its children, no need to even check where we are.
-                if (expectedException) {
-                    int startCurrentObjectName = -1;
-                    int endCurrentObjectName = -1;
-                    //look backwards for the current object name, to find out whether we expect an exception following its mutation
-                    for (int i = insertionPosition; i >= 0; i--) {
-                        if (validQuery.charAt(i) == '}') {
-                            break;
-                        } else if (validQuery.charAt(i) == '"') {
-                            if (endCurrentObjectName == -1) {
-                                endCurrentObjectName = i;
-                            } else if (startCurrentObjectName == -1) {
-                                startCurrentObjectName = i + 1;
-                            } else {
-                                break;
+    static Map<String, Boolean> alterateQueries(Set<String> queries, Set<String> arbitraryMarkers) throws IOException {
```

Comment:
I think either use `List<Tuple<String, Boolean>>` or a specially made object with javadocs to hold the results. Map is a bit weird to me here.